src/main/java/de/fraunhofer/aisec/cpg/passes/scopes/ScopeManager.java

```diff
@@ -364,80 +380,191 @@ public LabelStatement getLabelStatement(String labelString) {
+
+    Scope nameScope = resolveScopeWithPath(path);
+    if (nameScope instanceof NameScope) {
+      NamespaceDeclaration namespace = (NamespaceDeclaration) nameScope.getAstNode();
+      FunctionDeclaration existing =
+          namespace.getFunctions().stream()
+              .filter(containedField -> getSimpleName(containedField).equals(simpleName))
+              .collect(Collectors.toList())
+              .get(0);
+      if (existing != null) {
+        namespace.getFunctions().remove(existing);
+        namespace.getFunctions().add(function);
+      }
+    } else {
+      LOGGER.warn("Scope found with path to the field is not scoping a record declaration.");
     }
-    scopeForFunction.addValueDeclaration(functionDeclaration);
-    if (scopeForFunction.getAstNode() != null) {
-      RecordDeclaration rDecl = (RecordDeclaration) scopeForFunction.getAstNode();
-      List<FunctionDeclaration> functions = new ArrayList<>();
-      if (functionDeclaration instanceof ConstructorDeclaration) {
-        functions =
-            rDecl.getConstructors().stream()
-                .map(m -> (FunctionDeclaration) m)
-                .collect(Collectors.toList());
-      } else if (functionDeclaration instanceof MethodDeclaration) {
-        functions =
-            rDecl.getMethods().stream()
-                .map(m -> (FunctionDeclaration) m)
-                .collect(Collectors.toList());
+  }
+
+  public void addExternalRecordDeclaration(RecordDeclaration recordDeclaration) {}
+
+  // 1. enter scope of element to replace and move to parent or
+
+  public void addDeclaration(Declaration declaration) {
+
+    addDeclarationInCurrentScope(declaration);
+    return;
+    /*
+    if (!declaration.getName().contains(lang.getNamespaceDelimiter())) {
+      addDeclarationInCurrentScope(declaration);
+    } else {
+      Scope tmpScope = currentScope;
+      // Todo search for the right scope to add
+      // Todo set that scope as current scope
+      String prefix = declaration.getName();
+      prefix = prefix.substring(0, prefix.lastIndexOf(lang.getNamespaceDelimiter()));
+      Scope scope = resolveScopeWithPath(prefix);
+      if (scope != null) {
+        currentScope = scope;
+        addDeclarationInCurrentScope(declaration);
+      } else {
+        LOGGER.warn(
+            "External declaration {} could not be added to unknown parent {}.",
+            declaration.getName(),
+            prefix);
+        return;
       }
-      if (!functions.contains(functionDeclaration)) functions.add(functionDeclaration);
+
+      // Cut of the name prefix before adding to the appropriate scope
+      String name = declaration.getName();
+      name =
+          name.substring(
+              name.lastIndexOf(
+                  lang.getNamespaceDelimiter() + lang.getNamespaceDelimiter().length()));
+      declaration.setName(name);
+
+      addDeclarationInCurrentScope(declaration);
+      currentScope = tmpScope;
+    }
+    */
```

Comment:
I would rather just delete all of this commented out code

Reply:
Yes that was something I was not sure about because of me potentially needing it for future PRs but I guess I will keep my local branch where I still have these implementations and remove it for the sake of code quality.